pkg/handlers/generic/lifecycle/cni/calico/strategy_crs.go

```diff
@@ -208,12 +218,10 @@ func generateTigeraOperatorConfigMap(
 	return namespacedTigeraConfigMap
 }
 
-func generateProviderCNICRS(
+func generateProviderCNIManifestsConfigMap(
```

Comment:
Were the changes here something you found with the e2e tests?

Reply:
Kind of, but not precisely. When I first started using the owner reference e2e checks, I hit an error because we make the `Cluster` the owner of the `ClusterResourceSet` for GC. We also set up the `Cluster` as the owner of the referenced `ConfigMaps`. I fixed this up by having our own custom ownership checks for the `ClusterResourceSets` but had to remove the ownership from `ConfigMaps` as these should be owned by the `ClusterResourceSets` (applied by CRS controller). s part of that I noticed we could use one of our utility funcs to create the CRS rather than it being applied in each controller separately...

So in summary... these changes were discovered via the e2e implementation, but actually didn't show up ny huge problems with what we were doing, it's just a little tidier now.